### PR TITLE
csi: update default cephcsi version to 3.10.0

### LINKED ISSUE #13309: Integrate CephCSI v3.10 when the CSI release is out
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
For Rook v1.13, we need the Ceph-CSI v3.10 release integration.

**What is use case behind this feature:**
New features and fixes from CSI driver.

### PR DESCRIPTION
**Pull request by Rakshith-R**
This commit updates default cephcsi driver version to v3.10.0 and filesystem reconciler now creates
csi subvolumegroup by default.

**Which issue is resolved by this Pull Request:**
Resolves #13309 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/file/filesystem.go [hunk 1]
```diff
@@ -72,6 +73,11 @@ func createFilesystem(
 		}
 	}
 
+	err := cephclient.CreateCephFSSubVolumeGroup(context, clusterInfo, fs.Name, defaultCSISubvolumeGroup)
```
Comment (parth-gr) [inline pkg/operator/ceph/file/filesystem.go:76]: Also make sure of it's deletion. https://github.com/red-hat-storage/ocs-operator/pull/2271/files#diff-fabbfe52f64e79da27f9fff14bbd71e9a6bc122fd84073a8a1a7debc4aa9ac30R168

Comment (parth-gr) [inline pkg/operator/ceph/file/filesystem.go:76]: Can be a separate PR you can open, for testing and visibility

Comment (parth-gr) [inline pkg/operator/ceph/file/filesystem.go:76]: I think we should create the CR, instead of directly calling it, 
As we need to reconcile it + we need to call pinning + we have use new name spec if multiple filesystem is created(currently it also handled with your change).

**Review by parth-gr — CHANGES_REQUESTED** [review]

**Comment by Rakshith-R** [comment]
`ceph fs subvolumegroup create <fs_name> csi` is idempotent,
There's no need to associate this with a svg CR or pinning cmd. This can just exist within filesystem reconcile. 
There's no need for explicit deletion of default svg csi too. We never needed to delete it.

**Comment by parth-gr** [comment]
> `ceph fs subvolumegroup create <fs_name> csi` is idempotent, There's no need to associate this with a svg CR or pinning cmd. 

If you just create the CR, it will automatically pin to its default value, 

> This can just exist within filesystem reconcile. 

Yes we should keep this in filesystem reconcile

> There's no need for explicit deletion of default svg csi too. We never needed to delete it.

If the uninstallation code is called we need to remove it too with the filesystem

**Comment by Rakshith-R** [comment]
> > `ceph fs subvolumegroup create <fs_name> csi` is idempotent, There's no need to associate this with a svg CR or pinning cmd.
> 
> If you just create the CR, it will automatically pin to its default value,
> 

We can modify the flow when we have svg pinning available.  Currently, we need csi svg to be created to unlock cephcsi v3.10.0


> > This can just exist within filesystem reconcile.
> 
> Yes we should keep this in filesystem reconcile
> 
> > There's no need for explicit deletion of default svg csi too. We never needed to delete it.
> 
> If the uninstallation code is called we need to remove it too with the filesystem

Anyway the filesystem will be deleted, we don't need to worry about deleting svg explicitly.

**Comment by parth-gr** [comment]
> > > `ceph fs subvolumegroup create <fs_name> csi` is idempotent, There's no need to associate this with a svg CR or pinning cmd.
> > 
> > 
> > If you just create the CR, it will automatically pin to its default value,
> 
> We can modify the flow when we have svg pinning available. Currently, we need csi svg to be created to unlock cephcsi v3.10.0


It will be in probably by today https://github.com/rook/rook/pull/13283, @BlaineEXE 

> > > This can just exist within filesystem reconcile.
> > 
> > 
> > Yes we should keep this in filesystem reconcile
> > > There's no need for explicit deletion of default svg csi too. We never needed to delete it.
> > 
> > 
> > If the uninstallation code is called we need to remove it too with the filesystem
> 
> Anyway the filesystem will be deleted, we don't need to worry about deleting svg explicitly.

Then it will keep the traces than on the machine, right?

File: pkg/operator/ceph/csi/version.go [hunk 2]
```diff
@@ -25,14 +25,14 @@ import (
 )
 
 var (
-	//minimum supported version is 3.8.0
-	minimum = CephCSIVersion{3, 8, 0}
+	//minimum supported version is 3.9.0
+	minimum = CephCSIVersion{3, 9, 0}
 	//supportedCSIVersions are versions that rook supports
-	releasev390 = CephCSIVersion{3, 9, 0}
+	releasev3100 = CephCSIVersion{3, 10, 0}
```
Comment (travisn) [inline pkg/operator/ceph/csi/version.go:31]: How about dropping the patch from the var name? No need for the name to have it IMO.
```suggestion
	releasev310 = CephCSIVersion{3, 10, 0}
```

File: pkg/operator/ceph/file/filesystem_test.go [hunk 3]
```diff
@@ -142,6 +142,7 @@ func fsExecutor(t *testing.T, fsName, configDir string, multiFS bool, createData
 	if multiFS {
 		return &exectest.MockExecutor{
 			MockExecuteCommandWithOutput: func(command string, args ...string) (string, error) {
+				fmt.Println(args[0:5])
```
Comment (travisn) [inline pkg/operator/ceph/file/filesystem_test.go:145]: Was this just temporary? If keeping it, let's use the logger methods.
```suggestion
				logger.Info(args[0:5])
```

File: pkg/operator/ceph/file/filesystem.go (hunk 1 above)
Comment (travisn) [reply pkg/operator/ceph/file/filesystem.go:76]: I believe this was already resolved in the separate comments, but just to clarify... We will create svg here to ensure it exists, and the CR for the default svg will really just be needed for pinning settings, which would then be applied by that svg controller. Correct?

**Review by travisn — CHANGES_REQUESTED** [review]

**Review by travisn — COMMENTED** [review]
Are there no updates to the csi template manifests?

File: deploy/charts/rook-ceph/templates/role.yaml [hunk 4]
```diff
@@ -71,6 +71,9 @@ rules:
   - apiGroups: ["coordination.k8s.io"]
     resources: ["leases"]
     verbs: ["get", "watch", "list", "delete", "update", "create"]
+  - apiGroups: ["csiaddons.openshift.io"]
+    resources: ["csiaddonsnodes"]
+    verbs: ["create"]
```
[the remark this answers is not in the document]
Comment (Rakshith-R) [reply deploy/charts/rook-ceph/templates/role.yaml:76]: > no need for other operations like delete or update?

no, there's no need for update and deletion is taken care by owner ref of the csiaddonsnode object on the pod itself.

File: pkg/operator/ceph/file/filesystem_test.go (hunk 3 above)
Comment (Rakshith-R) [reply pkg/operator/ceph/file/filesystem_test.go]: it was for debugging, removed it.

**Comment by Rakshith-R** [comment]
> I believe this was already resolved in the separate comments, but just to clarify... We will create svg here to ensure it exists, and the CR for the default svg will really just be needed for pinning settings, which would then be applied by that svg controller. Correct?

Yes, That's correct.

> Are there no updates to the csi template manifests?

No, not for this release.

File: deploy/charts/rook-ceph/templates/clusterrole.yaml [hunk 5]
```diff
@@ -389,8 +389,6 @@ rules:
   - get
   - list
 ---
-# TODO: remove this, once https://github.com/rook/rook/issues/10141
```
Comment (Rakshith-R) [inline deploy/charts/rook-ceph/templates/clusterrole.yaml:392]: This clusterRole for getting nodes is now required for cephfs read affinity feature.
I've modified read affinity docs to indicate the same too.

**Review by travisn — APPROVED** [review]
